frame/node-authorization/src/lib.rs

```diff
@@ -386,23 +452,25 @@ pub mod pallet {
 		) -> DispatchResult {
 			let sender = ensure_signed(origin)?;
 
-			ensure!(node.0.len() < T::MaxPeerIdLength::get() as usize, Error::<T>::PeerIdTooLong);
+			let node = BoundedPeerId::<T>::try_from(node)?;
 			let owner = Owners::<T>::get(&node).ok_or(Error::<T>::NotClaimed)?;
 			ensure!(owner == sender, Error::<T>::NotOwner);
 
-			let mut nodes = AdditionalConnections::<T>::get(&node);
-
+			let mut nodes: BTreeSet<_> = AdditionalConnections::<T>::get(&node).into();
```

Comment:
Nit:
```suggestion
			let mut nodes = AdditionalConnections::<T>::get(&node);
```
then you can `try_insert` below and dont have to bound it again.